Automattic/wp-calypso
# Move i18n to an external npm package and update usage

This refactor should help us maintain and update our i18n code out of wp-calypso.
Also we want to reuse this library in other project and share it with the community.

For this PR, the following packages were created:
- `i18n-calypso` [repo](https://github.com/Automattic/i18n-calypso)
- `react-test-env` [repo](https://github.com/Automattic/react-test-env)

You might want to review them independently.

## Testing Instructions
Check that the following still works:
- Changing your locale
- Enabling the Community Translator and translating a sentence (does not seem to work locally, even on master...)
- Test all the following urls (as a logged out user):
    - http://calypso.localhost:3000/start/fr
    - http://calypso.localhost:3000/log-in/fr
    - http://calypso.localhost:3000/phone/fr
    - http://calypso.localhost:3000/jetpack/connect/fr

They should match the pages used in production

## Reviews

- [x] Product review v2


## Post merge tasks
- [ ] Fix the link to i18n documentation here: https://github.com/Automattic/eslint-plugin-wpcalypso/blob/master/docs/rules/i18n-no-variables.md 
- [ ] Use the I18N class from `i18n-calypso` to add internationalized SSR
- [ ] Port #5278 to `i18n-calypso`
